docs/WATSONX_EVALUATION_README.md

```diff
@@ -0,0 +1,243 @@
+# Watsonx Evaluation Integration for Maestro
+
+## Overview
+
+This document outlines the integration of IBM's watsonx governance evaluation capabilities into the Maestro multi-agent platform. The goal is to automatically evaluate agent responses for quality, safety, and reliability.
+
+We have successfully implemented **automatic evaluation middleware** that seamlessly integrates with existing Maestro agents without requiring any code changes. **The system is fully functional and returning actual metric scores!**
+
+### ✅ What's Working
+
+#### 1. **Automatic Evaluation Middleware**
+- **Zero Configuration**: Just set `MAESTRO_AUTO_EVALUATION=true` to enable
+- **Transparent Integration**: Works with any existing agent (`MockAgent`, `OpenAIAgent`, etc.)
+- **Production Ready**: Graceful fallback when watsonx library isn't available
+- **Clean Implementation**: Minimal debug output, production-ready code
+- **Real Metrics**: Returns actual numerical scores from watsonx evaluation
+
+#### 2. **Watsonx Integration**
+- **Decorator Pattern**: Successfully implemented IBM watsonx's decorator-based evaluation
+- **Authentication**: Integrated with IBM Cloud API authentication
+- **Multiple Metrics**: Evaluates:
+  - **Answer Relevance**: Measures how well the response addresses the question
+  - **Faithfulness**: Measures how faithful the response is to provided context
+  - **Context Relevance**: Measures how relevant the context is to the question
+  - **Answer Similarity**: Measures similarity between expected and actual answers
+- **Actual Scores**: Returns real numerical scores (0.0-1.0) with evaluation methods
+
+#### 2.1. **Available Metrics (Current)**
+
+| Metric              | Needs Context? | Needs Expected Answer? | Notes |
+|---------------------|----------------|-------------------------|-------|
+| Answer Relevance    | No             | No                      | Response addresses the question |
+| Faithfulness        | Yes            | No                      | Response grounded in provided context |
+| Context Relevance   | Yes            | No                      | Context relevance to the question |
+| Answer Similarity   | No             | Yes                     | Similarity to a reference answer |
+
+Inputs:
+- Context must be a list of strings (e.g., `["doc chunk 1", "doc chunk 2"]`).
+- Expected answer is a single string.
+- If a required input is missing, that metric is skipped gracefully.
+
+#### 3. **Environment Setup**
+- **Python 3.11**: Dedicated `.venv-eval` environment for watsonx compatibility
```

Comment:
is 3.11 a testing statement? We now have 3.12 in maestro also, with 3.13/3.14 expected
(As an optional extra, if 3.11 is a hard dependency due to third parties that's fine - may be worth elaborating)